lib/scalyr/common/client.rb

```diff
@@ -144,7 +144,7 @@ def close
   private
   def prepare_post_object(uri_path, body)
     # use compression if enabled
-    encoding = nil
+    encoding = "identity"
```

Comment:
If `@compression_type` is not set, encoding is not added to the`Content-Encoding` header in line 168.  Have you verified that the server will assume no compression if encoding is not set.  If that works, there's no need to set `encoding = "identity"` here because it is never used later.